simple-facebook-instant-articles.php

```diff
@@ -425,6 +427,35 @@ public function render_images( \DOMDocument &$dom, \DOMXPath &$xpath ) {
 	}
 
 	/**
+	 * Facebook throws a warning for all headings below h2.
+	 *
+	 * Replace with h2s.
+	 *
+	 * @param  \DOMDocument &$dom   DOM object generated for post content.
+	 * @param  \DOMXPath    &$xpath XPATH object generated for post content.
+	 *
+	 * @return void
+	 */
+	public function fix_headings( \DOMDocument &$dom, \DOMXPath &$xpath ) {
+
+		$headings = array( 'h3', 'h4', 'h5', 'h6' );
+
+		foreach ( $headings as $heading_tag ) {
+
+			foreach ( $dom->getElementsByTagName( $heading_tag ) as $node ) {
+
+				$h2 = $dom->createElement( 'h2' );
+
+				while ( $node->childNodes->length > 0 ) {
+					$h2->appendChild( $node->childNodes->item( 0 ) );
```

Comment:
Hm, isn't this gonna be infinite loop though? As we're not decreasing the childNodes? Also I still think this says append the fist child, rather than all.